src/tesk_core/callback_sender.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+import requests
+
+
+class CallbackSender:
+    """Implemention for the callback sender class."""
+
+    def __init__(self, task_id="", url=""):
+        self.url = url
+        self.task_id = task_id
+
+    def send(self, state):
+        """Send the state to the callback receiver.
+        Args:
+            state (str): Descriptor of the state of the task.
+        Returns:
+            response (requests.models.Response): Response from the callback receiver.
+            None: if the callback receiver is not set or some error occurs.
+        """
+
+        if not self.url:
+            return None
+        sent = False
```

Comment:
I guess instead of a `while` loop you could use a `for` loop with `range(number_of_retries)`. That way you could get rid of `retries` and manual incrementing. You could also use `break` instead of `sent = True` and get rid of `sent`.